File: back_API_ERP_Supermercado/src/main/java/com/example/back_API_ERP_Supermercado/Entidad/Personal.java
package com.example.back_API_ERP_Supermercado.Entidad;

import java.time.LocalDate;
import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "personal")
public class Personal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id_personal;

    private String nombre;
    private String apellido;
    private String carnet;

    @Column(name="fecha_creacion", nullable = true, updatable = false)
    private LocalDateTime fecha_creacion;

    @Column(name="id_rol")
    private Integer idRol;

    @Column(name="id_turno")
    private Integer idTurno;

    @Column(name="id_usuario")
    private long idUsuario;

    @PrePersist
    protected void onCreate() {
        fecha_creacion = LocalDateTime.now();
    }
}

File: front_ERP_Supermercado/src/interface/personal.ts
export interface personal {
    id_personal?: number;
    nombre: string;
    apellido: string;
    carnet: string;
    fecha_creacion: Date;
    idRol: number;
    idTurno: number;
    idUsuario: number;
}

export interface crearPersonal {
    id_personal?: number;
    nombre: string;
    apellido: string;
    carnet: string;
    fecha_creacion: Date;
    idRol: number;
    idTurno: number;
    username: string;
    email: string;
    password: string;
    idUsuario?: number;
}